temporalcli/commands.operator_namespace.go

```diff
@@ -0,0 +1,335 @@
+	nsName := args[0]
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	var updateRequest *workflowservice.UpdateNamespaceRequest
+
+	if c.PromoteGlobal {
+		fmt.Printf("Will promote local namespace to global namespace for:%s, other flag will be omitted. "+
+			"If it is already global namespace, this will be no-op.\n", nsName)
+		updateRequest = &workflowservice.UpdateNamespaceRequest{
+			Namespace:        nsName,
+			PromoteNamespace: true,
+		}
+	} else if len(c.ActiveCluster) > 0 {
+		fmt.Printf("Will set active cluster name to: %s, other flag will be omitted.\n", c.ActiveCluster)
+		replicationConfig := &replication.NamespaceReplicationConfig{
+			ActiveClusterName: c.ActiveCluster,
+		}
+		updateRequest = &workflowservice.UpdateNamespaceRequest{
+			Namespace:         nsName,
+			ReplicationConfig: replicationConfig,
+		}
+	} else {
+		resp, err := cl.WorkflowService().DescribeNamespace(cctx, &workflowservice.DescribeNamespaceRequest{
+			Namespace: nsName,
+		})
+		if err != nil {
+			switch err.(type) {
+			case *serviceerror.NamespaceNotFound:
+				return err
+			default:
+				return fmt.Errorf("namespace update failed: %w", err)
+			}
+		}
+
+		description := resp.NamespaceInfo.GetDescription()
+		ownerEmail := resp.NamespaceInfo.GetOwnerEmail()
+		retention := resp.Config.GetWorkflowExecutionRetentionTtl()
+
+		if len(c.Description) > 0 {
+			description = c.Description
+		}
+		if len(c.Email) > 0 {
+			ownerEmail = c.Email
+		}
+
+		data := map[string]string{}
+		if len(c.Data) > 0 {
+			data, err = stringKeysValues(c.Data)
+			if err != nil {
+				return err
+			}
+		}
+
+		if len(c.Retention) > 0 {
+			ret, err := timestamp.ParseDurationDefaultDays(c.Retention)
+			if err != nil {
+				return fmt.Errorf("option retention format is invalid: %w", err)
+			}
+			retention = durationpb.New(ret)
+		}
+
+		var clusters []*replication.ClusterReplicationConfig
+		if len(c.Cluster) > 0 {
+			for _, clusterName := range c.Cluster {
+				clusters = append(clusters, &replication.ClusterReplicationConfig{
+					ClusterName: clusterName,
+				})
+			}
+		}
+
+		updateInfo := &namespace.UpdateNamespaceInfo{
+			Description: description,
+			OwnerEmail:  ownerEmail,
+			Data:        data,
+		}
+
+		updateConfig := &namespace.NamespaceConfig{
+			WorkflowExecutionRetentionTtl: retention,
+			HistoryArchivalState:          archivalState(c.HistoryArchivalState.String()),
+			HistoryArchivalUri:            c.HistoryUri,
+			VisibilityArchivalState:       archivalState(c.VisibilityArchivalState.String()),
+			VisibilityArchivalUri:         c.VisibilityUri,
+		}
+		replicationConfig := &replication.NamespaceReplicationConfig{
+			Clusters: clusters,
+		}
+		updateRequest = &workflowservice.UpdateNamespaceRequest{
+			Namespace:         nsName,
+			UpdateInfo:        updateInfo,
+			Config:            updateConfig,
+			ReplicationConfig: replicationConfig,
+		}
+	}
+
+	if c.Verbose {
```

Comment:
I vote we break compat and remove `verbose` option for this call. If we want client request/response verbosity, we can do it at a higher level (i.e. via gRPC interceptor layer) and we'd use the logger not the printer.